engine/classes/ElggObject.php

```diff
@@ -150,13 +150,19 @@ protected function update() {
 		if (!parent::update()) {
 			return false;
 		}
+
+		update_river_access_by_object($this->guid, $this->access_id);
```

Comment:
AFAIK the `access_id` column in `river` table is basically obsolete. Need to verify this.